pd/src/genesis.rs

```diff
@@ -64,6 +127,50 @@ pub struct AppState {
     pub allocations: Vec<Allocation>,
 }
 
+impl From<AppState> for pb::GenesisAppState {
+    fn from(a: AppState) -> Self {
+        pb::GenesisAppState {
+            epoch_duration: a.epoch_duration,
+            // validators: a.validators.iter().map(|v| v.into()).collect(),
+            validators: a
+                .validators
+                .into_iter()
+                .map(TryInto::try_into)
+                .collect::<Result<_, _>>()
+                .expect("able to serialize validators"),
```

Comment:
Hmm, does it not work to do `.map(Into::into).collect()`? I think the domain types are supposed to be infallibly convertible into the proto types, does that not work here for some reason?